stl/inc/format

```diff
@@ -2199,19 +2195,21 @@ private:
     basic_format_args<basic_format_context> _Args;
     _Lazy_locale _Loc;
 
+    constexpr basic_format_context(_Out&& _OutputIt_, const basic_format_args<basic_format_context>& _Ctx_args)
+        : _OutputIt(_STD move(_OutputIt_)), _Args(_Ctx_args) {}
+
+    constexpr basic_format_context(
+        _Out&& _OutputIt_, const basic_format_args<basic_format_context>& _Ctx_args, const _Lazy_locale& _Loc_)
+        : _OutputIt(_STD move(_OutputIt_)), _Args(_Ctx_args), _Loc(_Loc_) {}
+
 public:
     using iterator  = _Out;
     using char_type = _CharT;
 
     template <class _Ty>
     using formatter_type = formatter<_Ty, _CharT>;
 
-    constexpr basic_format_context(_Out _OutputIt_, basic_format_args<basic_format_context> _Ctx_args)
-        : _OutputIt(_STD move(_OutputIt_)), _Args(_Ctx_args) {}
-
-    constexpr basic_format_context(
-        _Out _OutputIt_, basic_format_args<basic_format_context> _Ctx_args, const _Lazy_locale& _Loc_)
-        : _OutputIt(_STD move(_OutputIt_)), _Args(_Ctx_args), _Loc(_Loc_) {}
+    basic_format_context() = default;

```

Comment:
I'm not convinced we should add this - I think it's a specification oversight.

I observe that libstdc++ makes this default-constructible, but libc++ and fmtlib don't: https://godbolt.org/z/69qvc5E7M

Reply:
There's some weirdness in libstdc++ ([link](https://godbolt.org/z/x3v44PTbq), reported as [bug 114387](https://gcc.gnu.org/bugzilla/show_bug.cgi?id=114387)). It seems that there should be an LWG issue to clarify `basic_format_context`'s special member functions.